pkg/api/error.go

```diff
@@ -108,6 +108,18 @@ func NewCloudError(statusCode int, code, target, message string, a ...interface{
 	}
 }
 
+// NewInternalServerError returns a new CloudError with the internal server error standard text
+func NewInternalServerError(correlationData *CorrelationData) *CloudError {
```

Comment:
I don't think this is the right approach.  If you look at azure/cli/core/commands/arm.py, you'll see that it checks the response headers for x-ms-correlation-request-id, but we don't set this in pkg/frontend/middleware/log.go.  If we do, does it help?  While you're at it, we are also not compliant with https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#client-request-headers because we don't honour x-ms-return-client-request-id.  Please fix that at the same time?